crates/noirc_frontend/src/hir/def_collector/errors.rs

```diff
@@ -29,6 +30,19 @@ pub enum DefCollectorErrorKind {
     NonStructTypeInImpl { span: Span },
     #[error("Cannot `impl` a type defined outside the current crate")]
     ForeignImpl { span: Span, type_name: String },
+    #[error("Mismatch signature of trait")]
+    MismatchTraitSignature { primary_message: String, secondary_message: String, span: Span },
+    #[error("Method is not defined in trait")]
+    MethodNotInTrait {
+        trait_name: String,
+        trait_span: Span,
+        impl_method_name: String,
+        impl_method_span: Span,
+    },
+    #[error("Not a trait")]
+    NotATrait { primary_message: String, secondary_message: String, span: Span },
```

Comment:
Same here, lets avoid using `String` parameters for entire messages in general

Reply:
done